File: FSM/Entity/CustomerOrders/CustomerOrder.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.CustomerOrders.CustomerOrder
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using System.Collections;
using System.Runtime.CompilerServices;

namespace FSM.Entity.CustomerOrders
{
    public class CustomerOrder
    {
        private DataTypeValidator _dataTypeValidator;
        private bool _exists;
        private bool _deleted;
        private int _CustomerOrderID;
        private int _OrderID;
        private int _CustomerID;

        public CustomerOrder()
        {
            this._exists = false;
            this._deleted = false;
            this._CustomerOrderID = 0;
            this._OrderID = 0;
            this._CustomerID = 0;
            this._dataTypeValidator = new DataTypeValidator();
        }

        public bool IgnoreExceptionsOnSetMethods
        {
            get
            {
                return this._dataTypeValidator.IgnoreExceptionsOnSetMethods;
            }
            set
            {
                this._dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
            }
        }

        public Hashtable Errors
        {
            get
            {
                return this._dataTypeValidator.Errors;
            }
        }

        public DataTypeValidator DTValidator
        {
            get
            {
                return this._dataTypeValidator;
            }
        }

        public bool Exists
        {
            get
            {
                return this._exists;
            }
            set
            {
                this._exists = value;
            }
        }

        public bool Deleted
        {
            get
            {
                return this._deleted;
            }
        }

        public bool SetDeleted
        {
            set
            {
                this._deleted = value;
            }
        }

        public int CustomerOrderID
        {
            get
            {
                return this._CustomerOrderID;
            }
        }

        public object SetCustomerOrderID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_CustomerOrderID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int OrderID
        {
            get
            {
                return this._OrderID;
            }
        }

        public object SetOrderID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_OrderID", RuntimeHelpers.GetObjectValue(value));
            }
        }

        public int CustomerID
        {
            get
            {
                return this._CustomerID;
            }
        }

        public object SetCustomerID
        {
            set
            {
                this._dataTypeValidator.SetValue((object)this, "_CustomerID", RuntimeHelpers.GetObjectValue(value));
            }
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Order/CustomerOrder/CustomerOrder.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace CustomerOrders

Public Class CustomerOrder

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "CustomerOrder Properties"

      
Private _CustomerOrderID As Integer = 0
Public Readonly Property CustomerOrderID() As Integer
	Get 
		Return _CustomerOrderID
	End Get	
End Property	
Public Writeonly Property SetCustomerOrderID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_CustomerOrderID", Value)
	End Set
End Property


Private _OrderID As Integer = 0
Public Readonly Property OrderID() As Integer
	Get 
		Return _OrderID
	End Get	
End Property	
Public Writeonly Property SetOrderID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_OrderID", Value)
	End Set
End Property


Private _CustomerID As Integer = 0
Public Readonly Property CustomerID() As Integer
	Get 
		Return _CustomerID
	End Get	
End Property	
Public Writeonly Property SetCustomerID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_CustomerID", Value)
	End Set
End Property


  
#End Region

End Class

End Namespace

End Namespace

